src/vs/workbench/browser/parts/editor/editorDropTarget.ts

```diff
@@ -292,6 +296,31 @@ class DropOverlay extends Themable {
 			}
 		}
 
+		// Check for tree items
+		else if (this.treeItemsTransfer.hasData(DraggedTreeItemsIdentifier.prototype)) {
+			const data = this.treeItemsTransfer.getData(DraggedTreeItemsIdentifier.prototype);
+			if (Array.isArray(data)) {
+				const treeData = Promise.all(
+					data.map(id => this.treeViewsDragAndDropService.removeDragOperationTransfer(id.identifier)));
+				treeData.then(dataTransferItems => {
+					return Promise.all(dataTransferItems.map(async (dataTransferItem) => {
```

Comment:
There is still somewhat of a loop here right? I was wondering whether we should first collect all editors from all transfers and then open them all at once with a single `openEditors` call?

Reply:
Pushed a change that does not loop over `openEditors`.